### DISCUSSION TITLE
Bindings for API headers

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#3276] [Help]
Hi,
Is it possible to have binding for headers in API?
I would like to explore the possibility of binding the user JWT session in the header so that the API could get the logged in username to create audit log in the backend.

**Comment by ryzam** [comment]
Yes that's possible. This is an example step you can follow.

1. In your API configuration your need to set variable parameter and use that variable in json body or query string.
![image](https://user-images.githubusercontent.com/597759/140648823-9493d270-707d-483c-8647-38bd61e8e16d.png)

2. Then assign the value from your Form using "Execute Query" when you click Define Action
![image](https://user-images.githubusercontent.com/597759/140649089-112dd96b-8613-4fe7-9159-eb1e9f292fd7.png)

Hope this can help

**Reply by micebrain** [reply]
Thank @ryzam for the example.
I am actually looking for binding in the header parameters not the request body. 
Something like

<img width="648" alt="Screenshot 2021-11-08 at 20 21 37" src="https://user-images.githubusercontent.com/7136209/140804540-6cadbc63-2a1c-4520-be28-ed404fc89798.png">

**Reply by mjashanks** [reply]
What you have done here should work - should it not?

**Reply by micebrain** [reply]
Unfortunately no. In fact in Action - Execute Query, the binding option did not appear.

![Screenshot 2021-11-09 at 08 40 55](https://user-images.githubusercontent.com/7136209/140882847-83f80d45-c443-45e1-b69c-193e1ca38176.png)

**Reply by mjashanks** [reply]
Ah yea, there is now binding drawer in that interface. You have to type it manually.... `{{ yourParameterName }}`

**Reply by micebrain** [reply]
Yes, tried with manual entry. But my backend api gives me `{{ yourParameterName }}` instead of the bind value... 
Looks like a bug then.

**Reply by mjashanks** [reply]
Just got around to trying to reproduce this. It's working ok for me. I setup a local API and debuged the headers....

![image](https://user-images.githubusercontent.com/3524181/141337486-b9c630ec-189e-4e08-ae1a-d6fa364d7964.png)

Then I called it via a data provider - my supplied parameter came through.

**Reply by mjashanks** [reply]
I just realised that in your screenshot, you are trying to bind the headers directly in the Design section. I can see why - it's confusing because they are text inputs. But those aren't editable.

You need to 
1. Setup a parameter on your datasource (as per my screenshot)
2. Bind the parameter in the data source (as per my screenshot)
3. In the design section, bind the parameter to something in your UI

**Reply by micebrain** [reply]
Oh! Okay, I didn't thought that it's supposed to be a parameter :) 
Thanks for the tips!
